### PR TITLE
Gridicons: All gridicons should be stored in the `UIImage+Woo` extension

### LINKED ISSUE #746: Gridicons: All gridicons should be stored in the `UIImage+Woo` extension
In order to keep images tidy and add a bit of optimization, let's find all `Gridicon.iconOfType` uses found in other classes and replace them.

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #746 

Encapsulate all access to GridIcons in the UIImage+Woo extension. Considering the announcements at WWDC, this seemed like the right moment to do this 😄 

## Changes
- Added vars to the images extension, and reordered the existing ones in alphabetical order (sorry about that 😊 )
- Added tests to ensure the integrity of the dependency

## Testing
- Checkout the branch, run the tests. The tests should pass
- Search the codebase for `Gridicon.iconOfType`. All occurrences should be only in `UIImage+Woo` (and a few results in Pods, out of the scope of this story)
- Navigate around, check that icons are still rendered and that I haven't messed up anything

### CODE REVIEW

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift [hunk 1]
```diff
@@ -21,17 +53,107 @@ extension UIImage {
...
+        return Gridicon.iconOfType(.chevronDown)
+    }
+
+    /// Chevron pointing up
+    ///
+    static var chevronUpImage: UIImage {
+        return Gridicon.iconOfType(.chevronUp)
+    }
+
+    /// Cog image
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/UIImage+Woo.swift]: `Cog Icon`

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift [hunk 2]
```diff
@@ -21,17 +53,107 @@ extension UIImage {
         return Gridicon.iconOfType(.chevronRight).imageWithTintColor(tintColor)!
     }
 
+    /// Chevron pointing down
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/UIImage+Woo.swift]: To keep the comments consistent, would you Title Case all the icon and image names? This one would be `Chevron Pointing Down`

File: WooCommerce/Classes/ViewRelated/Products/Cells/ProductReviewsTableViewCell.swift [hunk 3]
```diff
@@ -44,12 +44,10 @@ final class ProductReviewsTableViewCell: UITableViewCell {
 //
 private extension ProductReviewsTableViewCell {
     enum Star {
-        static let size = CGSize(width: 20, height: 20)
-        static let filledImage = Gridicon.iconOfType(.star,
-                                                     withSize: Star.size
-            ).imageWithTintColor(StyleManager.grayStarColor)
-        static let emptyImage = Gridicon.iconOfType(.starOutline,
```
Comment (mindgraffiti) [inline WooCommerce/Classes/ViewRelated/Products/Cells/ProductReviewsTableViewCell.swift]: .starOutline icon is missing

File: WooCommerce/Classes/ViewRelated/Products/Cells/ProductReviewsTableViewCell.swift [hunk 4]
```diff
@@ -44,12 +44,10 @@ final class ProductReviewsTableViewCell: UITableViewCell {
...
-                                                    withSize: Star.size
-            ).imageWithTintColor(StyleManager.grayStarColor)
+        static let size = Double(20)
+        static let filledImage = UIImage.starImage(size: Star.size,
+                                                   tintColor: StyleManager.grayStarColor)
+        static let emptyImage = UIImage.starImage(size: Star.size,
```
Comment (mindgraffiti) [inline WooCommerce/Classes/ViewRelated/Products/Cells/ProductReviewsTableViewCell.swift]: This should be a `UIImage.starOutlineImage`

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift [hunk 5]
```diff
@@ -43,28 +165,64 @@ extension UIImage {
...
+    ///   - tintColor: desired tint color of the resulting icon
+    /// - Returns: a bitmap image
+    ///
+    static func starImage(size: Double, tintColor: UIColor) -> UIImage {
+        let starSize = CGSize(width: size, height: size)
+        return Gridicon.iconOfType(.star,
+                                   withSize: starSize)
+            .imageWithTintColor(tintColor)!
+    }
+
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/UIImage+Woo.swift]: Right after the `starImage`, insert the `starOutlineImage`.

`Gridicon.iconOfType(.starOutline)` then it needs to be added in `ProductDetailsViewController`.

**Comment by ctarda** [comment]
Thanks @mindgraffiti and sorry I missed that icon 😬

This is ready for another look

**Pull request merged**